tests/AtlasSearchTest.php

```diff
@@ -194,9 +203,62 @@ public function testDatabaseBuilderAutocomplete()
         self::assertInstanceOf(LaravelCollection::class, $results);
         self::assertCount(3, $results);
         self::assertSame([
-            'Operating System Concepts',
             'Database System Concepts',
             'Modern Operating Systems',
-        ], $results->all());
+            'Operating System Concepts',
+        ], $results->sort()->values()->all());
+    }
+
+    public function testDatabaseBuilderVectorSearch()
+    {
+        $results = $this->getConnection('mongodb')->table('books')
+            ->vectorSearch(
+                index: 'vector',
+                path: 'vector4',
+                queryVector: $this->vectors[7], // This is an exact match of the vector
+                limit: 4,
+                exact: true,
+            );
+
+        self::assertInstanceOf(LaravelCollection::class, $results);
+        self::assertCount(4, $results);
+        self::assertSame('The Art of Computer Programming', $results->first()['title']);
+        self::assertSame(1.0, $results->first()['vectorSearchScore']);
+    }
+
+    public function testEloquentBuilderVectorSearch()
+    {
+        $results = Book::vectorSearch(
+            index: 'vector',
+            path: 'vector4',
+            queryVector: $this->vectors[7],
+            limit: 5,
+            numCandidates: 15,
+            // excludes the exact match
+            filter: Query::query(
+                title: Query::ne('The Art of Computer Programming'),
+            ),
+        );
+
+        self::assertInstanceOf(EloquentCollection::class, $results);
+        self::assertCount(5, $results);
+        self::assertInstanceOf(Book::class, $results->first());
+        self::assertNotSame('The Art of Computer Programming', $results->first()->title);
+        self::assertSame('The Mythical Man-Month: Essays on Software Engineering', $results->first()->title);
+        self::assertThat(
+            $results->first()->vectorSearchScore,
+            self::logicalAnd(self::isType('float'), self::greaterThan(0.9), self::lessThan(1.0)),
+        );
+    }
+
+    /** Generate random vectors */
+    private function addVector(array $items): array
+    {
+        srand(1);
+        foreach ($items as &$item) {
+            $this->vectors[] = $item['vector4'] = array_map(fn () => rand() / mt_getrandmax(), range(0, 3));
```

Comment:
Consider leaving a TODO to revisit this once [PHPC-2474](https://jira.mongodb.org/browse/PHPC-2474) is implemented. It looks like that ticket entails more than just adding a new subtype constant, as there are utility functions to convert to/from vectors (as described in [the corresponding spec](https://github.com/mongodb/specifications/blob/master/source/bson-binary-vector/bson-binary-vector.md#api-guidance)).

Reply:
Not sure for the todo, I create a ticket to add support in the aggregation builder first. https://jira.mongodb.org/browse/PHPLIB-1603